ansible/roles/operator-pipeline/templates/openshift/tasks/get-ci-reviewer.yml

```
@@ -0,0 +1,79 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: get-ci-reviewer
+spec:
+  params:
+    - name: pipeline_image
+    - name: git_username
+    - name: operator_name
+    - name: pr_url
+    - name: github_token_secret_key
+      description: The key within the Kubernetes Secret that contains the GitHub token.
+      default: token
+    - name: github_token_secret_name
+      description: The name of the Kubernetes Secret that contains the GitHub token.
+      default: github
+  workspaces:
+    - name: repo
+      description: Cloned repository
+  results:
+    - name: author_is_reviewer
+      description: >-
+        Check if the author of the PR is listed also as a reviewer
+    - name: list_reviewers
+      description: >-
+        Return list of all reviewers from ci.yaml file except author of the pull request
+  steps:
+    - name: get-ci-reviewer
+      image: "$(params.pipeline_image)"
+      workingDir: $(workspaces.repo.path)
+      script: |
+        #! /usr/bin/env bash
+        set -e
+        CI="operators/$(params.operator_name)/ci.yaml"
+        if [ ! -e "$CI" ]; then
+          echo >$(results.list_reviewers.path)
+          echo false >$(results.author_is_reviewer.path)
+        else
+          if grep -q 'reviewers:' $CI; then
+            yq -r '.reviewers[]' $CI >$(results.list_reviewers.path)
+            yq -r ".reviewers|index(\"$(params.git_username)\") != null" $CI >$(results.author_is_reviewer.path)
+          else
+            echo >$(results.list_reviewers.path)
+            echo false >$(results.author_is_reviewer.path)
+          fi
```

Would be more reliable and easier to read this way:
```suggestion
          yq -r '(.reviewers//[]))[]' $CI >$(results.list_reviewers.path)
          yq -r ".reviewers//[]|index(\"$(params.git_username)\") != null" $CI >$(results.author_is_reviewer.path)
```

---

I tried the suggestion (with some additional changes trying to make it work) but it does not work because when the reviewers key is not there, the indexing in yq would fail - was searching for that issue yesterday/few days ago and it has something to do with how content from yaml is then parsed to json during storing it in results and if key is not there, it just fails on the indexing, if i remember correctly. 
I had the same issue as in the previous iteration, that is the reason why i switched to greping the key directly and then going through the additional else condition. This way i made it work and i did not have any errors  - whether reviewers are missing or the entire ci.yaml file is missing.